python-modules/cis_crypto/cis_crypto/operation.py

```diff
@@ -73,20 +82,63 @@ def _get_public_key(self):
                     )
                 )
             )
-            key_name = self.config('public_key_name', namespace='cis')
+            key_name = self.config('public_key_name', namespace='cis', default='access-file-key')
             file_name = '{}.pub.pem'.format(key_name)
             fh = open((os.path.join(key_dir, file_name)), 'rb')
             key_content = fh.read()
             key_construct = jwk.construct(key_content, 'RS256')
             return key_construct.to_dict()
-        elif self.well_known_mode is 'http':
+        elif self.well_known_mode == 'http' or self.well_known_mode == 'https':
             # Go get it from the .well-known endpoint and load as json
             # return a dictionary of the json loaded data
-            pass
+            well_known_url = self.config('well_known_url', namespace='cis')
+            res = requests.get(well_known_url)
+
+            if res.status_code == 200:
+                key_list = self._reduce_keys(res.json())
+                return key_list
```

Comment:
Maybe return `False` if the well-known fetch fails? This just returns nothing, not sure how your upstream code deals with this method returning nothing (maybe setting a variable to the return value of a method that doesn't return sets it to `None`, I'm note sure)